src/figuresClasses.ts

```diff
@@ -1,11 +1,97 @@
-export interface Figure {}
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+function getRoundedNumber(number: number): number {
+  return Math.floor(number * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  public shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a <= 0 ||
+      this.b <= 0 ||
+      this.c <= 0 ||
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.a + this.b
+    ) {
+      throw new Error('Sides must be positive and not equal to zero');
+    } else {
+    }
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const triSqr = Math.sqrt(p * ((p - this.a) * (p - this.b) * (p - this.c)));
+
+    return getRoundedNumber(triSqr);
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Circle implements Figure {
+  public shape = Shape.circle;
+
+  constructor(
+    public color: Color,
+    public r: number,
+  ) {
+    if (this.r <= 0) {
+      throw new Error('Radius must be positive and not equal to zero');
```

Comment:
The error message should be more descriptive. Consider specifying that the radius must be a positive number greater than zero.